src/ser.rs

```diff
@@ -48,9 +49,10 @@ impl<'a, 'b, 'c, S> VariantSerializer<'a, 'b, 'c, S> {
   fn end(self, inner: impl FnOnce(S) -> JsResult<'a>) -> JsResult<'a> {
     let value = inner(self.inner)?;
     let scope = &mut *self.scope.borrow_mut();
-    let obj = v8::Object::new(scope);
+    let null = v8::null(scope).into();
     let key = v8_struct_key(scope, self.variant).into();
-    obj.set(scope, key, value);
+    let obj =
+      v8::Object::with_prototype_and_properties(scope, null, &[key], &[value]);
```

Comment:
I was concerned about deserializing to objects with null prototype, but the [`no-prototype-builtins`](https://github.com/tc39/test262/blob/main/test/built-ins/Array/of/does-not-use-prototype-properties.js) lint rule should prevent any side effects.